pre-commit is not happy with latest and stable

As seen in a recent PR, pre-commit is not happy with `latest` and `stable` as rev: values.

Happy to submit a PR, what version number do you want to use?

Which version of black? - 20.8b1 is latest
https://github.com/psf/black/releases

Which version of reuse tool?  - v0.12.1 is latest
https://github.com/fsfe/reuse-tool/releases

Another option would be to use an Organizational secret like I'm sure you're doing with AWS auth, etc... Which would fix maintaining these values,  but would not run locally for contributors.

------
Example of the failure:

https://github.com/adafruit/Adafruit_CircuitPython_DisplayIO_SH1107/runs/1788637904?check_suite_focus=true#step:11:1
```
Warning:  The 'rev' field of repo 'https://github.com/python/black' appears to be a mutable reference
(moving tag / branch).  
Mutable references are never updated after first install and are not supported. 
See https://pre-commit.com/#using-the-latest-version-for-a-repository for more details.

Warning:  The 'rev' field of repo 'https://github.com/fsfe/reuse-tool' appears to be a mutable reference
(moving tag / branch). 
Mutable references are never updated after first install and are not supported. 
See https://pre-commit.com/#using-the-latest-version-for-a-repository for more details.

[INFO] Initializing environment for https://github.com/python/black.
An unexpected error has occurred: CalledProcessError: command: ('/usr/bin/git', 'checkout', 'latest')
return code: 1
expected return code: 0
stdout: (none)
stderr:
    error: pathspec 'latest' did not match any file(s) known to git
    
Check the log at /home/runner/.cache/pre-commit/pre-commit.log
Error: Process completed with exit code 3.
```


------------

This approach of hard-coding the specific version is problematic to manage, so I'm going to test several things and see if we can come up with another approach.

- Using Secrets to specify a version for a pre-commit hook. Let's pretend this would work from within GitHub. Locally, it fails 100% so its a non-starter. Secrets don't appear to be available outside Actions as well.
- using [command line options](https://pre-commit.com/#pre-commit-autoupdate), NOPE this is not available as a modifier to `run`
  ~- `autoupdate`~
  ~- `autoupdate --bleeding-edge`~

## Centralize pre-commit.yaml

~- PR to pre-commit to support --config and .pre-commit.yaml referring to an externally hosted yaml file (URL based) which could be hosted in a GIST or a repo.~ - https://github.com/pre-commit/pre-commit/issues/450 Discusses the need for this and the maintainer saying. "No."
  - [SO discussion about that](https://stackoverflow.com/questions/59047996/how-can-i-configure-pre-commit-tool-to-rely-on-a-single-location-for-all-hook-co) approach by having one command in your pre-commit being a custom hook that gets the config you want and then calls pre-commit. Note: That is from the pre-commit maintainer.
- Someone wrote a tool to pull configs and then insrtall pre-commit, not the same as the wrapper hook: [Pierre-Sassoulas/centralized-pre-commit-conf](https://github.com/Pierre-Sassoulas/centralized-pre-commit-conf) The readme is pretty good. ;-)

## Update pre-commit.yaml everywhere

Maintaining .pre-commit.yaml separately and when it changes update all affected repos. Some ways to do that:
  - Kattni mentioned [Adabot](https://github.com/adafruit/adabot/pull/204/files)
  - I found [Gitbot](https://github.com/Clever/gitbot)
  - Maintainer of pre-commit also publishes [all-repos](https://github.com/asottile/all-repos#example-autofixer) and in the above bug they mention this as a way to manage this


I do think Adafruit is going to need a way to update across repos no matter what. Rolling your own or looking into Gitbot or All-Repos is my suggestion with a heavy leaning towards _not_ rolling your own. ;-)

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -4,7 +4,7 @@
 
 repos:
 -   repo: https://github.com/fsfe/reuse-tool
-    rev: latest
+    rev: v0.12.1
     hooks:
     - id: reuse
 -   repo: https://github.com/pre-commit/pre-commit-hooks

--- a/{{ cookiecutter.library_name }}/.pre-commit-config.yaml
+++ b/{{ cookiecutter.library_name }}/.pre-commit-config.yaml
@@ -4,11 +4,11 @@
 
 repos:
 -   repo: https://github.com/python/black
-    rev: stable
+    rev: 20.8b1
     hooks:
     - id: black
 -   repo: https://github.com/fsfe/reuse-tool
-    rev: latest
+    rev: v0.12.1
     hooks:
     - id: reuse
 -   repo: https://github.com/pre-commit/pre-commit-hooks

